### ISSUE TITLE
The given NumPy array is not writable, and PyTorch does not support non-writable tensors.

### ISSUE THREAD

**Issue by xyzjsj** [NVlabs/OmniDrive#70]
`2025-05-23 20:11:43,940 - mmdet - INFO - workflow: [('train', 1)], max: 21096 iters
2025-05-23 20:11:43,945 - mmdet - INFO - Checkpoints will be saved to /home/jiangyanbo/OmniDrive/work_dirs/mask_eva_lane_det_vlm by HardDiskBackend.
Token indices sequence length is longer than the specified maximum sequence length for this model (2696 > 2048). Running this sequence through the model will result in indexing errors
Token indices sequence length is longer than the specified maximum sequence length for this model (2288 > 2048). Running this sequence through the model will result in indexing errors
/home/jiangyanbo/OmniDrive/projects/mmdet3d_plugin/models/utils/grid_mask.py:113: UserWarning: The given NumPy array is not writable, and PyTorch does not support non-writable tensors. This means writing to this tensor will result in undefined behavior. You may want to copy the array to protect its data or make it writable before converting it to a tensor. This type of warning will be suppressed for the rest of this program. (Triggered internally at ../torch/csrc/utils/tensor_numpy.cpp:199.)
  mask = torch.from_numpy(mask).float().cuda()
/home/jiangyanbo/miniconda3/envs/omnidrive/lib/python3.9/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3190.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
Token indices sequence length is longer than the specified maximum sequence length for this model (2062 > 2048). Running this sequence through the model will result in indexing errors
Token indices sequence length is longer than the specified maximum sequence length for this model (2429 > 2048). Running this sequence through the model will result in indexing errors
Token indices sequence length is longer than the specified maximum sequence length for this model (2298 > 2048). Running this sequence through the model will result in indexing errors
Token indices sequence length is longer than the specified maximum sequence length for this model (2458 > 2048). Running this sequence through the model will result in indexing errors
/home/jiangyanbo/OmniDrive/projects/mmdet3d_plugin/models/utils/grid_mask.py:113: UserWarning: The given NumPy array is not writable, and PyTorch does not support non-writable tensors. This means writing to this tensor will result in undefined behavior. You may want to copy the array to protect its data or make it writable before converting it to a tensor. This type of warning will be suppressed for the rest of this program. (Triggered internally at ../torch/csrc/utils/tensor_numpy.cpp:199.)
  mask = torch.from_numpy(mask).float().cuda()
/home/jiangyanbo/OmniDrive/projects/mmdet3d_plugin/models/utils/grid_mask.py:113: UserWarning: The given NumPy array is not writable, and PyTorch does not support non-writable tensors. This means writing to this tensor will result in undefined behavior. You may want to copy the array to protect its data or make it writable before converting it to a tensor. This type of warning will be suppressed for the rest of this program. (Triggered internally at ../torch/csrc/utils/tensor_numpy.cpp:199.)
  mask = torch.from_numpy(mask).float().cuda()
/home/jiangyanbo/OmniDrive/projects/mmdet3d_plugin/models/utils/grid_mask.py:113: UserWarning: The given NumPy array is not writable, and PyTorch does not support non-writable tensors. This means writing to this tensor will result in undefined behavior. You may want to copy the array to protect its data or make it writable before converting it to a tensor. This type of warning will be suppressed for the rest of this program. (Triggered internally at ../torch/csrc/utils/tensor_numpy.cpp:199.)
  mask = torch.from_numpy(mask).float().cuda()
Token indices sequence length is longer than the specified maximum sequence length for this model (2400 > 2048). Running this sequence through the model will result in indexing errors
Token indices sequence length is longer than the specified maximum sequence length for this model (2164 > 2048). Running this sequence through the model will result in indexing errors
/home/jiangyanbo/miniconda3/envs/omnidrive/lib/python3.9/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3190.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
/home/jiangyanbo/miniconda3/envs/omnidrive/lib/python3.9/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3190.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
/home/jiangyanbo/miniconda3/envs/omnidrive/lib/python3.9/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3190.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
2025-05-23 20:12:06,832 - mmcv - INFO - Reducer buckets have been rebuilt in this iteration.
2025-05-23 20:20:47,048 - mmdet - INFO - Iter [50/21096]        lr: 1.595e-04, eta: 2 days, 15:28:04, time: 10.856, data_time: 0.054, memory: 32215, loss_cls: 2.2350, loss_bbox: 3.0224, d0.loss_cls: 2.4011, d0.loss_bbox: 3.0050, d1.loss_cls: 2.1718, d1.loss_bbox: 3.0102, d2.loss_cls: 2.1505, d2.loss_bbox: 2.9965, d3.loss_cls: 2.2245, d3.loss_bbox: 2.9944, d4.loss_cls: 2.2897, d4.loss_bbox: 2.9948, dn_loss_cls: 2.2118, dn_loss_bbox: 2.8369, d0.dn_loss_cls: 2.3741, d0.dn_loss_bbox: 2.7510, d1.dn_loss_cls: 2.1437, d1.dn_loss_bbox: 2.3825, d2.dn_loss_cls: 2.1222, d2.dn_loss_bbox: 2.3025, d3.dn_loss_cls: 2.1960, d3.dn_loss_bbox: 2.5076, d4.dn_loss_cls: 2.2599, d4.dn_loss_bbox: 2.8284, loss_cls_lane: 1.4441, loss_cls_H: 1.4606, loss_bbox_lane: 8.9624, loss_bbox_H: 8.8305, d0.loss_cls_lane: 1.5190, d0.loss_cls_H: 1.4988, d0.loss_bbox_lane: 8.9896, d0.loss_bbox_H: 8.9135, d1.loss_cls_lane: 1.5239, d1.loss_cls_H: 1.5277, d1.loss_bbox_lane: 9.2294, d1.loss_bbox_H: 9.1172, d2.loss_cls_lane: 1.5528, d2.loss_cls_H: 1.5505, d2.loss_bbox_lane: 9.5658, d2.loss_bbox_H: 9.4355, d3.loss_cls_lane: 1.5538, d3.loss_cls_H: 1.5592, d3.loss_bbox_lane: 9.8999, d3.loss_bbox_H: 9.7725, d4.loss_cls_lane: 1.5166, d4.loss_cls_H: 1.5159, d4.loss_bbox_lane: 10.0669, d4.loss_bbox_H: 9.9107, vlm_loss: 0.4941, loss: 191.8236, grad_norm: nan`

UserWarning: The given NumPy array is not writable, and PyTorch does not support non-writable tensors. This means writing to this tensor will result in undefined behavior. You may want to copy the array to protect its data or make it writable before converting it to a tensor. This type of warning will be suppressed for the rest of this program. 

Is it normal？ I also see it in [issue63](https://github.com/NVlabs/OmniDrive/issues/63)

Here is my environment
`Package                   Version        Editable project location
------------------------- -------------- ----------------------------------------------------
absl-py                   2.2.2
accelerate                0.29.0
addict                    2.4.0
anyio                     4.9.0
argon2-cffi               23.1.0
argon2-cffi-bindings      21.2.0
arrow                     1.3.0
asttokens                 3.0.0
async-lru                 2.0.5
attrs                     25.3.0
babel                     2.17.0
beautifulsoup4            4.13.4
black                     25.1.0
bleach                    6.2.0
cachetools                5.5.2
certifi                   2025.4.26
cffi                      1.17.1
chardet                   5.2.0
charset-normalizer        3.4.2
click                     8.1.8
colorama                  0.4.6
comm                      0.2.2
contourpy                 1.3.0
cycler                    0.12.1
debugpy                   1.8.14
decorator                 5.2.1
defusedxml                0.7.1
descartes                 1.1.0
einops                    0.8.1
exceptiongroup            1.3.0
executing                 2.2.0
fastjsonschema            2.21.1
filelock                  3.18.0
fire                      0.7.0
flake8                    7.2.0
flash-attn                2.5.6
fonttools                 4.58.0
fqdn                      1.5.1
fsspec                    2025.5.0
grpcio                    1.71.0
h11                       0.16.0
httpcore                  1.0.9
httpx                     0.28.1
huggingface-hub           0.31.4
idna                      3.10
imageio                   2.37.0
importlib_metadata        8.7.0
importlib_resources       6.5.2
iniconfig                 2.1.0
ipykernel                 6.29.5
ipython                   8.18.1
ipywidgets                8.1.7
iso3166                   2.1.1
isoduration               20.11.0
jedi                      0.19.2
Jinja2                    3.1.6
joblib                    1.5.0
json5                     0.12.0
jsonpointer               3.0.0
jsonschema                4.23.0
jsonschema-specifications 2025.4.1
jupyter                   1.1.1
jupyter_client            8.6.3
jupyter-console           6.6.3
jupyter_core              5.7.2
jupyter-events            0.12.0
jupyter-lsp               2.2.5
jupyter_server            2.16.0
jupyter_server_terminals  0.5.3
jupyterlab                4.4.2
jupyterlab_pygments       0.3.0
jupyterlab_server         2.27.3
jupyterlab_widgets        3.0.15
kiwisolver                1.4.7
llvmlite                  0.36.0
lyft-dataset-sdk          0.0.8
Markdown                  3.8
markdown-it-py            3.0.0
MarkupSafe                3.0.2
matplotlib                3.5.3
matplotlib-inline         0.1.7
mccabe                    0.7.0
mdurl                     0.1.2
mistune                   3.1.3
mmcls                     0.25.0
mmcv-full                 1.6.2
mmdet                     2.28.2
mmdet3d                   1.0.0rc6       /home/jiangyanbo/OmniDrive/mmdetection3d
mmsegmentation            0.30.0
model-index               0.1.11
mpmath                    1.3.0
mypy_extensions           1.1.0
narwhals                  1.40.0
nbclient                  0.10.2
nbconvert                 7.16.6
nbformat                  5.10.4
nest-asyncio              1.6.0
networkx                  3.2.1
ninja                     1.11.1.4
notebook                  7.4.2
notebook_shim             0.2.4
numba                     0.53.0
numpy                     1.23.4
nuscenes-devkit           1.1.11
nvidia-cublas-cu12        12.6.4.1
nvidia-cuda-cupti-cu12    12.6.80
nvidia-cuda-nvrtc-cu12    12.6.77
nvidia-cuda-runtime-cu12  12.6.77
nvidia-cudnn-cu12         9.5.1.17
nvidia-cufft-cu12         11.3.0.4
nvidia-cufile-cu12        1.11.1.6
nvidia-curand-cu12        10.3.7.77
nvidia-cusolver-cu12      11.7.1.2
nvidia-cusparse-cu12      12.5.4.2
nvidia-cusparselt-cu12    0.6.3
nvidia-nccl-cu12          2.26.2
nvidia-nvjitlink-cu12     12.6.85
nvidia-nvtx-cu12          12.6.77
opencv-python             4.11.0.86
opendatalab               0.0.10
openlanev2                2.1.0          /home/jiangyanbo/OmniDrive/mmdetection3d/OpenLane-V2
openmim                   0.3.9
openxlab                  0.0.11
ordered-set               4.1.0
ortools                   9.2.9972
overrides                 7.7.0
packaging                 25.0
pandas                    2.2.3
pandocfilters             1.5.1
parso                     0.8.4
pathspec                  0.12.1
peft                      0.4.0
pexpect                   4.9.0
pillow                    11.2.1
pip                       25.1.1
platformdirs              4.3.8
plotly                    6.1.1
pluggy                    1.6.0
plyfile                   1.1
prettytable               3.16.0
prometheus_client         0.22.0
prompt_toolkit            3.0.51
protobuf                  6.31.0
psutil                    7.0.0
ptyprocess                0.7.0
pure_eval                 0.2.3
pycocotools               2.0.8
pycodestyle               2.13.0
pycparser                 2.22
pycryptodome              3.23.0
pyflakes                  3.3.2
Pygments                  2.19.1
pyparsing                 3.2.3
pyquaternion              0.9.9
pytest                    8.3.5
python-dateutil           2.9.0.post0
python-json-logger        3.3.0
pytz                      2025.2
PyWavelets                1.6.0
PyYAML                    6.0.2
pyzmq                     26.4.0
referencing               0.36.2
regex                     2024.11.6
requests                  2.32.3
rfc3339-validator         0.1.4
rfc3986-validator         0.1.1
rich                      14.0.0
rpds-py                   0.25.1
safetensors               0.5.3
scikit-image              0.19.3
scikit-learn              1.6.1
scipy                     1.8.0
Send2Trash                1.8.3
sentencepiece             0.2.0
setuptools                59.5.0
Shapely                   1.8.5.post1
similaritymeasures        1.2.0
six                       1.17.0
sniffio                   1.3.1
soupsieve                 2.7
stack-data                0.6.3
sympy                     1.14.0
tabulate                  0.9.0
tensorboard               2.19.0
tensorboard-data-server   0.7.2
termcolor                 3.1.0
terminado                 0.18.1
terminaltables            3.1.10
threadpoolctl             3.6.0
tifffile                  2024.8.30
tinycss2                  1.4.0
tokenizers                0.13.3
tomli                     2.2.1
torch                     1.13.1+cu117
torchaudio                0.13.1+cu117
torchvision               0.14.1+cu117
tornado                   6.5
tqdm                      4.67.1
traitlets                 5.14.3
transformers              4.31.0
trimesh                   2.35.39
triton                    3.3.0
types-python-dateutil     2.9.0.20250516
typing_extensions         4.13.2
tzdata                    2025.2
uri-template              1.3.0
urllib3                   2.4.0
wcwidth                   0.2.13
webcolors                 24.11.1
webencodings              0.5.1
websocket-client          1.8.0
Werkzeug                  3.1.3
wheel                     0.45.1
widgetsnbextension        4.0.14
yapf                      0.32.0
zipp                      3.21.0`

**Comment by exiawsh** [comment]
Hi,
It's normal.
